src/base/settingsstorage.h

```diff
@@ -45,18 +49,48 @@ class SettingsStorage : public QObject
     static void freeInstance();
     static SettingsStorage *instance();
 
-    QVariant loadValue(const QString &key, const QVariant &defaultValue = {}) const;
-    void storeValue(const QString &key, const QVariant &value);
+    template<typename T>
+    void store(const QString &key, const T &value)
```

Comment:
Unlike `SettingValue` class (that represents some value) appropriate methods of this class should have `value` in their names to be unambiguous. You don't store/load `SettingsStorage` itself. You really call it to store/load some value.

Reply:
Reverted.